test/span_test.cpp

```diff
@@ -196,6 +196,42 @@ TEST_CASE("span") {
     REQUIRE(result->type == "new type");
   }
 
+  SECTION("error tag sets error") {
+    auto span_id = get_id();
+    Span span{nullptr,    buffer,  get_time, sampler,
+              span_id,    span_id, 0,        SpanContext{span_id, span_id, {}},
+              get_time(), "",      "",       "",
+              "",         ""};
+
+    struct ErrorTagTestCase {
+      ot::Value value;
+      uint32_t span_error;
+      std::string span_tag;
+    };
+
+    auto error_tag_test_case = GENERATE(values<ErrorTagTestCase>({
```

Comment:
I'm a fan of this.

Reply:
I like how it automatically makes matrix tests if you have two GENERATES :D 

It's really hard to not write tests like I used to in Golang land (Goland?), where it's a big loop over a bunch of TestCase structs.